File: app/src/main/java/com/alexpaxom/workrhythm/ui/main/ListTasksAdapter.java
package com.alexpaxom.workrhythm.ui.main;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.MutableLiveData;
import androidx.recyclerview.widget.RecyclerView;

import com.alexpaxom.workrhythm.R;
import com.alexpaxom.workrhythm.model.Task;

import org.jetbrains.annotations.NotNull;

import java.util.Collections;
import java.util.GregorianCalendar;
import java.util.List;


public class ListTasksAdapter extends RecyclerView.Adapter<ListTasksHolder> {


    MutableLiveData<Long> currTime_live;
    Fragment parentFragment;
    List<Task> data = Collections.emptyList();
    Integer shown_status;
    Thread timer_thread;

    public ListTasksAdapter(Fragment parentFragment, Integer shown_status) {
        this.parentFragment = parentFragment;
        this.shown_status = shown_status;

        if(shown_status == Task.IN_WORK)
            updateTimeInTasks();
    }

    private void updateTimeInTasks() {
        currTime_live = new MutableLiveData<>();
        currTime_live.setValue(GregorianCalendar.getInstance().getTimeInMillis());


        timer_thread = new Thread(() -> {
            while(!Thread.interrupted()) {
                try {
                    Thread.sleep(1000);
                    currTime_live.postValue(GregorianCalendar.getInstance().getTimeInMillis());
                } catch(InterruptedException ex) {
                    Thread.currentThread().interrupt();
                }
            }
        });

        timer_thread.start();
    }

    @NonNull
    @NotNull
    @Override
    public ListTasksHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        Context context = parent.getContext();

        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.item_task, parent, false);


        return new ListTasksHolder(this, view);
    }

    @Override
    public void onBindViewHolder(@NonNull @NotNull ListTasksHolder holder, int position) {
          holder.bind(position);
          if(shown_status == Task.IN_WORK)
            holder.updateTimeInTask();
    }

    public void reloadListData(List<Task> data) {
        this.data = data;
        notifyDataSetChanged();
    }


    @Override
    public int getItemCount() {
        return data.size();
    }

    public void onDestroyAdapter() {
        if(timer_thread != null && !timer_thread.isInterrupted())
            timer_thread.interrupt();
    }

}

File: app/src/main/java/com/alexpaxom/workrhythm/app/adapters/ListTasksAdapter.kt
package com.alexpaxom.workrhythm.app.adapters

import androidx.recyclerview.widget.RecyclerView
import com.alexpaxom.workrhythm.app.adapters.holders.ListTasksHolder
import androidx.lifecycle.MutableLiveData
import android.view.ViewGroup
import android.view.LayoutInflater
import androidx.fragment.app.Fragment
import com.alexpaxom.workrhythm.R
import com.alexpaxom.workrhythm.data.model.Task
import java.util.*

class ListTasksAdapter(var parentFragment: Fragment, var shown_status: Int) :
    RecyclerView.Adapter<ListTasksHolder>() {
    @JvmField
    var currTime_live: MutableLiveData<Long>? = null
    @JvmField
    var data: List<Task> = emptyList()
    var timer_thread: Thread? = null
    private fun updateTimeInTasks() {
        currTime_live = MutableLiveData()
        currTime_live!!.value = GregorianCalendar.getInstance().timeInMillis
        timer_thread = Thread {
            while (!Thread.interrupted()) {
                try {
                    Thread.sleep(1000)
                    currTime_live!!.postValue(GregorianCalendar.getInstance().timeInMillis)
                } catch (ex: InterruptedException) {
                    Thread.currentThread().interrupt()
                }
            }
        }
        timer_thread!!.start()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListTasksHolder {
        val context = parent.context
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(R.layout.item_task, parent, false)
        return ListTasksHolder(this, view)
    }

    override fun onBindViewHolder(holder: ListTasksHolder, position: Int) {
        holder.bind(position)
        if (shown_status === Task.IN_WORK) holder.updateTimeInTask()
    }

    fun reloadListData(data: List<Task>) {
        this.data = data
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return data.size
    }

    fun onDestroyAdapter() {
        if (timer_thread != null && !timer_thread!!.isInterrupted) timer_thread!!.interrupt()
    }

    init {
        if (shown_status === Task.IN_WORK) updateTimeInTasks()
    }
}